### PR TITLE
feat: support multi-project projects by adding PREFIX

### LINKED ISSUE #149: Error trying to have multiple projects that use project_options
There's a decent chance this is pure user error and I don't understand how to compose project_options properly, but I'm getting configure-time errors when trying to have multiple projects using project_options, specifically:

```
[cmake] CMake Error at build/_deps/_project_options-src/src/Index.cmake:125 (add_library):
[cmake]   add_library cannot create target "project_options" because another target
[cmake]   with the same name already exists.  The existing target is an interface
[cmake]   library created in source directory
[cmake]   "/home/mcoding/Projects/project_options_multiple_include_mwe".  See
[cmake]   documentation for policy CMP0002 for more details.
```

The project structure is like this:

```
project/
    apps/
        CMakeLists.txt
        main.cpp
    CMakeLists.txt
    external/mylib/
        CMakeLists.txt
        include/mylib
            mylib.hpp
```

Both `project` and the submodule `mylib` use `project_options`, which explains why cmake thinks `project_options` is being multiply defined, because both `project` and `mylib` do indeed define their own set of project options. Is this a situation where `project_options` should be defining target names that depend on the project name like `${PROJECT_NAME}_project_options` instead of a global name `project_options`? Or is there some way to have `project_options` scoped to a namespace like `mylib::project_options`? Or should only the top-level project be defining project options (using an `if TARGET` guard before including it) with subprojects modifying the options using `dynamic_project_options`?

Any guidance here would be appreciated.

Here are the relevant list files for the minimal working example:

Top level `CMakeLists.txt`:
```
cmake_minimum_required(VERSION 3.16)

include(FetchContent)
FetchContent_Declare(_project_options URL https://github.com/aminya/project_options/archive/refs/tags/v0.24.1.zip)
FetchContent_MakeAvailable(_project_options)
include(${_project_options_SOURCE_DIR}/Index.cmake)
include(${_project_options_SOURCE_DIR}/src/DynamicProjectOptions.cmake)

project(myproject LANGUAGES CXX C)

dynamic_project_options()

add_subdirectory(external/mylib)

add_subdirectory(apps)
```

`apps/CMakeLists.txt`
```
add_executable(main main.cpp)
target_link_libraries(main PRIVATE project_options project_warnings)

target_link_system_libraries(
  main
  PRIVATE mylib
)
```

`external/mylib/CMakeLists.txt`
```
cmake_minimum_required(VERSION 3.16)

include(FetchContent)
FetchContent_Declare(_project_options URL https://github.com/aminya/project_options/archive/refs/tags/v0.24.1.zip)
FetchContent_MakeAvailable(_project_options)
include(${_project_options_SOURCE_DIR}/Index.cmake)
include(${_project_options_SOURCE_DIR}/src/DynamicProjectOptions.cmake)

project(mylib LANGUAGES CXX C)

dynamic_project_options()

add_library(mylib INTERFACE)
target_link_libraries(mylib INTERFACE project_options project_warnings)

set(INCLUDE_DIR "include") 
target_include_directories(mylib INTERFACE "$<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/${INCLUDE_DIR}>"
                                          "$<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}>")
```

### PR DESCRIPTION
**Pull request by aminya**
Fixes #149

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -221,6 +221,7 @@ It accepts the following named flags:
 
 It gets the following named parameters that can have different values in front of them:
 
+- `PREFIX`: the optional prefix that is used to define `${PREFIX}_project_options` and `${PREFIX}_project_warnings` targets when the function is used in a multi-project fashion.
```
Comment (ClausKlein) [inline README.md:224]: **IMHO**: For may reasons, this should **NOT** optional -> **it must be mandatory!**

1. konsider: project A is installed (with exported `project_options`) and later, project B, which use also  `project_options`) depends from project A. It use find_package() and import  `project_options`
2. Or Project D depends from both: A and C, but does not use  `project_options`. But A and C use independent.

One solution for this future desaster is to never export  `project_options` and  `project_warnings`!
They must be used `PRVATE `only and never exported.

Comment (ClausKlein) [reply README.md:224]: **It is not a good idea to export static-analyser settings or gcov options!**

Comment (aminya) [reply README.md:224]: The default example only enables the static analyzers when the `FEATURE_TESTS ` is enabled

https://github.com/aminya/project_options#usage

Comment (ClausKlein) [reply README.md:224]: OK, I allways disable install in developer mode!

Comment (aminya) [reply README.md:224]: Do not turn off the installation. Just put the test and analyzer under an option. This is very common even without this library

Comment (aminya) [reply README.md:224]: I don't know which package manager you use, but the install rules are required for vcpkg and conan to work without any issues.

**Pull request merged**